rollodex/src/App.js

```diff
@@ -1,26 +1,98 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, {Component} from 'react';
+// import logo from './logo.svg';
 import './App.css';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isHidden: true, contacts: [],
+    }
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  fetchData() {
+    fetch ('https://randomuser.me/api?results=25')
+    .then(response => response.json())
+    .then(parsedJSON => parsedJSON.results.map(contact => ({
+      name: `${contact.name.first} ${contact.name.last}`,
+      location: `${contact.location.city}, ${contact.location.state}, ${contact.location.country}`,
+      email: `${contact.email}`,
+      cell: `${contact.cell}`,
+      username: `${contact.login.username}`,
+      mediumpic: `${contact.picture.medium}`,
+      smallpic: `${contact.picture.thumbnail}`
+    })))
+    .then(contacts => this.setState({
+      contacts,
+      isHidden: true
+    }))
+    .catch(error => console.log("parsing failed", error))
+  }
+
+  handleClick = () => {
+    let status = this.state.isHidden === true ? false : true
+    this.setState({
+      isHidden: status
+    })
+  }
+
+  render() {
+    const {isHidden, contacts} = this.state;
+    if (isHidden) {
+      return (
+        <div>
+          <header><h1>My Peeps</h1></header>
+          <div>
+            <div className="body-container">
+              {
+                contacts.map(contact => {
+                  const {name, username, smallpic} = contact;
+                  return (
+                    <div key={username} title={name} className="list-container">
+                      <button onClick={this.handleClick}>Show Details</button><br></br>
+                      <img src={smallpic} alt="thumbnail"></img>
+                      <p>{name}</p>
+                      <br></br>
+                    </div>
+                  )
+                })
+              }
```

Comment:
The use of a full function block (with requisite `return`) inside your `map` (or any `{...}` interpolation in JSX) is what I was recommending against in class.  Not a big deal (and assigning a couple variables for simpler access like this is a common rationale).  I suggest:
```
<div className="body-container">
  { contacts.map( contact => (
    <div key={contact.username} title={contact.name} className="list-container">
      <button onClick={this.handleClick}>Show Details</button><br></br>
      <img src={contact.smallpic} alt="thumbnail"></img>
      <p>{name}</p>
      <br></br>
    </div>
  ))}
</div>  
```
If you the `contact.*` references get to be too much of a pain you can always do this:
```
<div className="body-container">
  { contacts.map( contact => (
    <ContactCard contact={contact} onToggle={this.handleClick} index={index} />
  ))}
</div>
```
If you toggled by `id` you wouldn't need to pass the `index`

Reply:
Oh, forgot to *use* `isHidden`:
```
<div className="body-container">
  { contacts.map( contact => (
    <div key={contact.username} title={contact.name} className="list-container">
      <button onClick={this.handleClick}>Show Details</button><br></br>
      { contact.isHidden ? (
        <>
          <img src={contact.smallpic} alt="thumbnail"></img>
          <p>{contact.name}</p>
          <br></br>
        </>
      ):(
        <>
          <img src={contact.smallpic} alt="thumbnail"></img>
          <p>{contact.name}</p>
          <p>{contact.location}</p>
          /* all the details */
        </>
      )}
    </div>
  ))}
</div>
```

Also, FWIW you can use `<>` and `</>` in place of the leading & trailing parens there & it's a bit more succinct:
```
<div>
  { contact.isHidden ? <>
    <img />
    <p>
  </>:<>
    <img />
    <p>
  </>}
</div>
```
I know that's a lot of variation so if it's overwhelming forget that last bit... or even go the route of creating a sub-component.  That almost always makes things more readable, at the expense of having to jump around a lot.